openforcefield/tests/test_utils_structure.py

```diff
@@ -13,74 +13,107 @@
 # GLOBAL IMPORTS
 #=============================================================================================
 
-from functools import partial
-from unittest import TestCase
+import os
+
 import parmed
+import pytest
+
+from openforcefield.typing.engines.smirnoff import ForceField
+from openforcefield.topology import Molecule, Topology
 from openforcefield import utils
+import openforcefield.utils.structure as structure
+
 
 #=============================================================================================
 # TESTS
 #=============================================================================================
 
-class TestUtils(TestCase):
+requires_openeye_mol2 = pytest.mark.skipif(not utils.OpenEyeToolkitWrapper.is_available(),
+                                               reason='OpenEye is required to parse mol2 files')
+
+class TestUtilsStructure:
+
+    @requires_openeye_mol2
     def test_read_molecules(self):
-        molecules = utils.read_molecules('zinc-subset-tripos.mol2.gz', verbose=False)
+        molecules = structure.read_molecules('zinc-subset-tripos.mol2.gz', verbose=False)
+
+    @requires_openeye_mol2
     def test_positions(self):
         """Test ability to extract and set positions."""
-        molecules = utils.read_molecules('zinc-subset-tripos.mol2.gz', verbose=False)
-        positions = utils.extractPositionsFromOEMol(molecules[0])
-        utils.setPositionsInOEMol(molecules[0], positions)
+        molecules = structure.read_molecules('zinc-subset-tripos.mol2.gz', verbose=False)
+        positions = structure.extractPositionsFromOEMol(molecules[0])
+        structure.setPositionsInOEMol(molecules[0], positions)
+
+    @requires_openeye_mol2
     def test_smirnoff_structure(self):
-       """Test function that generates the SMIRNOFF parmed.Structure."""
-       molecule = utils.read_molecules('toluene.pdb', verbose=False)[0]
-       molecule_structure = utils.generateSMIRNOFFStructure(molecule)
-       self.assertIsInstance(molecule_structure, parmed.structure.Structure)
+        """Test function that generates the SMIRNOFF parmed.Structure."""
+        molecule = structure.read_molecules('toluene.pdb', verbose=False)[0]
+        molecule_structure = structure.generateSMIRNOFFStructure(molecule)
+        assert isinstance(molecule_structure, parmed.structure.Structure)
+
     def test_protein_structure(self):
         from simtk.openmm import app
         pdbfile = utils.get_data_filename('proteins/T4-protein.pdb')
         proteinpdb = app.PDBFile(pdbfile)
-        protein_structure = utils.generateProteinStructure(proteinpdb)
-        self.assertIsInstance(protein_structure, parmed.structure.Structure)
+        protein_structure = structure.generateProteinStructure(proteinpdb)
+        assert isinstance(protein_structure, parmed.structure.Structure)
+
 
+@requires_openeye_mol2
 def test_merge_system():
```

Comment:
Now that we're setting things like electrostatics cutoffs in the SMIRNOFF file, we will want to expand these tests to include what happens when systems with different forcefield parameters like that are combined. This shouldn't be done in this PR -- just a note to myself to open an issue.